src/utils/calculateReClammInitAmounts.ts

```diff
@@ -0,0 +1,91 @@
+import {
+    type InputAmount,
+    type MinimalToken,
+    isSameAddress,
+    MathSol,
+    getSortedTokens,
+    Token,
+    TokenAmount,
+} from 'src';
+import { computeInitialBalanceRatio } from '@balancer-labs/balancer-maths';
+
+export async function calculateReClammInitAmounts({
+    chainId,
+    tokens,
+    givenAmountIn,
+    initialMinPrice,
+    initialMaxPrice,
+    initialTargetPrice,
+}: {
+    tokens: MinimalToken[];
+    chainId: number;
+    givenAmountIn: InputAmount;
+    initialMinPrice: bigint;
+    initialMaxPrice: bigint;
+    initialTargetPrice: bigint;
+}): Promise<InputAmount[]> {
+    // sort the user provided tokens
+    const sortedTokens = getSortedTokens(tokens, chainId);
+    const givenTokenIndex = sortedTokens.findIndex((t) =>
+        isSameAddress(t.address, givenAmountIn.address),
+    );
+
+    const proportion = computeInitialBalanceRatio(
+        initialMinPrice,
+        initialMaxPrice,
+        initialTargetPrice,
+    );
+
+    let calculatedAmountIn: InputAmount;
+
+    const givenToken = new Token(
+        chainId,
+        givenAmountIn.address,
+        givenAmountIn.decimals,
+    );
+    const givenTokenAmount = TokenAmount.fromRawAmount(
+        givenToken,
+        givenAmountIn.rawAmount,
+    );
+    const givenTokenAmountScaled18 = givenTokenAmount.scale18;
+
+    // https://github.com/balancer/reclamm/blob/8207b33c1ab76de3c42b015bab5210a8436376de/test/reClammPool.test.ts#L120-L128
+    if (givenTokenIndex === 0) {
+        // if given token is first in sort order, we multiply (math must be done in scaled 18)
+        const rawAmountScaled18 = MathSol.mulDownFixed(
+            givenTokenAmountScaled18,
+            proportion,
+        );
+        // then convert back to token decimals raw amount (rounding down)
+        const rawAmountScaledNative = TokenAmount.fromScale18Amount(
+            givenToken,
```

Comment:
Found one issue.
This should be the `calculatedToken` instead of `givenToken`.
Same goes when givenTokenIndex is 1.
I won't be able to apply the fix to validate, but you can try making the change to see if tests pass.

Reply:
![there-goes-my-hero-foo-fighters](https://github.com/user-attachments/assets/6698108d-3eae-43f1-89d6-a81b443c5902)